scripts/upload-spotlight.js

```diff
@@ -0,0 +1,267 @@
+  add_help: true,
+});
+
+argparse.add_argument("-i", "--id", {
+  required: true,
+  help: "ID of dataset for collection",
+});
+
+argparse.add_argument("-s", "--ageadjusted", {
+  required: false, // TO_REVIEW should this be required?
+  help: "New data for the outcome stats",
+});
+
+argparse.add_argument("-a", "--agespecific", {
+  required: false, // TO_REVIEW should this be required?
+  help: "New data for the affordability outcomes",
+});
+
+argparse.add_argument("-o", "--overwrite", {
+  action: "store_true",
+  help: "if files already exist, overwrite them",
+});
+
+argparse.add_argument("-n", "--newId", {
+  action: "store_true",
+  help: "if the collection id does not exist, creates a new collection",
+});
+
+argparse.add_argument("-z", "--localDir", {
+  help: "path to local folder to save download files to",
+});
+
+const main = async () => {
+  const { id, ageadjusted, agespecific, newId, localDir } =
+    argparse.parse_args();
+
+  const files = [
+    {
+      filePath: ageadjusted,
+      extension: "json",
+      field: "adjusted", // TO_REVIEW what is field? used for localdir, is this the field in the firestore db?
+      isArray: true,
+      schema: [
+        {
+          name: "housing_type",
+          type: "string",
+        },
+        {
+          name: "age_adjusted_rate",
+          type: "number",
+        },
+        {
+          name: "outcome_type",
+          type: "string",
+        },
+      ],
+    },
+    {
+      filePath: agespecific,
+      extension: "json",
+      field: "specific",
+      isArray: true,
+      schema: [
+        {
+          name: "hud_age_group",
+          type: "string",
+        },
+        {
+          name: "final_housing_type",
+          type: "string",
+        },
+        {
+          name: "age_specific_rate",
+          type: "number",
+        },
+        {
+          name: "outcome_type",
+          type: "string",
+        },
+      ],
+    },
+  ];
+
+  files.forEach(({ filePath, extension }) => {
+    if (!fs.existsSync(filePath)) {
+      warnAndExit(`ERROR! File does not exist: ${filePath}`);
+    }
+
+    if (path.extname(filePath).toLowerCase() !== `.${extension}`) {
+      warnAndExit(`ERROR! Expected a .${extension} file: ${filePath}`);
+    }
+  });
+
+  process.env.GOOGLE_APPLICATION_CREDENTIALS = "serviceAccount.json";
+  const app = initializeApp();
+  const db = getFirestore(app);
+
+  // Check if the collection exists
+  const collections = (await db.listCollections()).map((c) => c.id);
+  if (!collections.includes(id)) {
+    if (newId) {
+      console.warn(
+        `WARNING! id does not exist in firestore. This script will create the following collection: ${id}`
+      );
+    } else {
+      warnAndExit(`ERROR! id must match an existing collection id: ${collections.join(
+        ", "
+      )}.
+      Use the --newId flag to create a new collection id for "${id}"`);
+    }
+  }
+
+  const docRef = db.collection(id);
```

Comment:
So this looks like it's just to a collection. This should be to a document to save the data in

Reply:
`const docRef = db.collection("spotlights").doc("housing")`